ksqldb-engine/src/main/java/io/confluent/ksql/engine/EngineExecutor.java

```diff
@@ -347,28 +369,128 @@ TransientQueryMetadata executeTransientQuery(
+  private KsqlPlan sourceTablePlan(
+      final ConfiguredStatement<?> statement) {
+    final CreateTable createTable = (CreateTable) statement.getStatement();
+    final CreateTableCommand ddlCommand = (CreateTableCommand) engineContext.createDdlCommand(
+        statement.getStatementText(),
+        (ExecutableDdlStatement) statement.getStatement(),
+        config
+    );
+
+    final Relation from = new AliasedRelation(
+        new Table(createTable.getName()), createTable.getName());
+
+    // Only VALUE columns must be selected from the source table. When running a pull query, the
+    // keys are added if selecting all columns.
+    final Select select = new Select(
+        createTable.getElements().stream()
+            .filter(column -> column.getNamespace() == TableElement.Namespace.VALUE)
+            .map(column -> new SingleColumn(
+                new UnqualifiedColumnReferenceExp(column.getName()),
+                Optional.of(column.getName())))
+            .collect(Collectors.toList()));
+
+    // Source table need to keep emitting changes so every new record is materialized for
+    // pull query availability.
+    final RefinementInfo refinementInfo = RefinementInfo.of(OutputRefinement.CHANGES);
+
+    // This is a plan for a `select * from <source-table> emit changes` statement,
+    // without a sink topic to write the results. The query is just made to materialize the
+    // source table.
+    final Query query = new Query(
+        Optional.empty(),
+        select,
+        from,
+        Optional.empty(),
+        Optional.empty(),
+        Optional.empty(),
+        Optional.empty(),
+        Optional.empty(),
+        Optional.of(refinementInfo),
+        false,
+        OptionalInt.empty());
+
+    // The source table does not exist in the current metastore, so a temporary metastore that
+    // contains only the source table is created here. This metastore is used later to create
+    // ExecutorsPlan.
+    final MutableMetaStore tempMetastore = new MetaStoreImpl(new InternalFunctionRegistry());
+    final Formats formats = ddlCommand.getFormats();
+    tempMetastore.putSource(new KsqlTable<>(
+        statement.getStatementText(),
+        createTable.getName(),
+        ddlCommand.getSchema(),
+        Optional.empty(),
+        false,
+        new KsqlTopic(
+            ddlCommand.getTopicName(),
+            KeyFormat.of(formats.getKeyFormat(), formats.getKeyFeatures(), Optional.empty()),
+            ValueFormat.of(formats.getValueFormat(), formats.getValueFeatures())
+        ),
+        true
+    ), false);
+
+    final ExecutorPlans plans = planQuery(
+        statement,
+        query,
+        Optional.empty(),
+        Optional.empty(),
+        tempMetastore
+    );
+
+    final KsqlBareOutputNode outputNode =
+        (KsqlBareOutputNode) plans.logicalPlan.getNode().get();
+
+    final QueryPlan queryPlan = new QueryPlan(
+        getSourceNames(outputNode),
+        null,
```

Comment:
The `QueryPlan` requires a `SourceName` for the sink topic. However, CST plans do not have sink topics. Passing an empty source name also fails because source names validate they're not empty. I couldn't find another way to avoid passing a null.

Reply:
Sure, approved